packages/playwright-core/src/web/traceViewer/ui/modelUtil.ts

```diff
@@ -16,14 +16,48 @@
 
 import { ResourceSnapshot } from '../../../server/trace/common/snapshotTypes';
 import { ActionTraceEvent } from '../../../server/trace/common/traceEvents';
-import { ContextEntry, MergedContexts, PageEntry } from '../entries';
+import { ContextEntry, PageEntry } from '../entries';
+import * as trace from '../../../server/trace/common/traceEvents';
 
 const contextSymbol = Symbol('context');
 const nextSymbol = Symbol('next');
 const eventsSymbol = Symbol('events');
 const resourcesSymbol = Symbol('resources');
 
-export function indexModel(context: ContextEntry) {
+export class MultiTraceModel {
```

Comment:
I'd move all exported functions in this file into MultiTraceModel. Do you plan to follow up?

Reply:
I didn't quite like that the callsites would become more verbose, e.g. `context(action).events.` => `MultiTraceModel.context(action).events.` It'd be better to convert/wrap ActionTraceEvent into a class with corresponding methods, wdyt?